discovery-infra/test_infra/helper_classes/kube_helpers/agent_cluster_install.py

```diff
@@ -268,6 +272,8 @@ def _has_required_condition() -> Optional[bool]:
                 if required_reason:
                     return required_reason == reason
                 return True
+            elif status == exception_status and reason == exception_reason:
+                raise Exception(f"Waiting for condition <{cond_type}> to be in status <{exception_status}>. Actual status is: {status} {exception_reason}")
```

Comment:
Save the above string in a variable and use it both for the log and for the exception message.

Reply:
rephrased message to make it clearer.